lib/teams/index.ts

```diff
@@ -294,3 +298,185 @@ export class PlatformTeam extends ApplicationTeam {
         this.defaultSetupAdminAccess(clusterInfo);
     }
 }
+
+/**
+ * Interface define the object to create an execution role
+ */
+export interface ExcutionRoleDefinition {
+    /**
+     * The name of the IAM role to create
+     */
+  excutionRoleName: string,
+  /**
+    * The IAM policy to use with IAM role if it already exists
+    * Can be initialized for example by `fromPolicyName` in Policy class
+    */
+  excutionRoleIamPolicy?: IManagedPolicy,
+  /**
+    * Takes an array of IAM Policy Statement, you should pass this 
+    * if you want the Team to create the policy along the IAM role 
+    */
+  excutionRoleIamPolicyStatement?: PolicyStatement[],
+}
+
+/**
+ * Interface to define a EMR on EKS team
+ */
+export interface EmrEksTeamProps extends TeamProps {
+  /*
+  * The namespace of where the virtual cluster will be created
+  */ 
+  virtualClusterNamespace: string,
+  /**
+   * To define if the namespace that team will use need to be created
+   */
+  createNamespace: boolean,
+  /*
+  * The name of the virtual cluster the team will use
+  */ 
+  virtualClusterName: string,
+  /**
+   * List of execution role to associated with the VC namespace
+   */
+  excutionRoles: ExcutionRoleDefinition[]
+}
+
+/*
+ *This class define the Team that can be used with EMR on EKS
+ *The class will create an EMR on EKS Virtual Cluster to use by the team
+ *It can either create a namespace or use an existing one
+ *The class will set the necessary k8s RBAC needed by EMR on EKS as defined in the AWS documentation 
+ * https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up-cluster-access.html
+ * The class constructor take a the EMR on EKS Team definition
+ */
+
+export class EmrEksTeam extends ApplicationTeam {
```

Comment:
Does it have to derive from the AppicationTeam? Is there any code reuse?

Reply:
Yes it need to extend `ApplicationTeam`, this can allow a user to provide access to IAM role owned for example to the namespace used by data engineers. It is useful to debug jobs or deployment and need access to logs/pods.